Remove pandas(-stubs) dependencies

daniel-zullo-frequenz opened the pull request:
The pandas(-stubs) dependencies are no longer needed.

The package numpy was imported through pandas so it needs to be added to the dependency list once pandas get removed.

daniel-zullo-frequenz committed:
Remove pandas(-stubs) dependencies

The pandas(-stubs) dependency are no longer neeeded.
The package numpy was imported through pandas, thus
it is added to the dependencies now that pandas is
removed.

Signed-off-by: Daniel Zullo <daniel.zullo@frequenz.com>

leandro-lucarella-frequenz reviewed (changes requested):
Please also remove `pandas` from the general dependencies (line 27) :)

daniel-zullo-frequenz commented:
> Please also remove pandas from the general dependencies (line 27) :)

I tried to remove it but it seems we still have dependencies on pandas itself. I'll need to investigate further

daniel-zullo-frequenz commented:
it seems numpy was imported through pandas as when I removed pandas the modules related to numpy cannot be found

daniel-zullo-frequenz committed:
Replace pytz with timezone module

This is done for handling UTC timezone without importing
the module pytz which is currently installed through pandas.

Signed-off-by: Daniel Zullo <daniel.zullo@frequenz.com>

daniel-zullo-frequenz commented:
Replaced pytz with timezone module for handling UTC timezone without importing the module pytz which was also installed through pandas.

ela-kotulska-frequenz commented:
Could you update release notes?

daniel-zullo-frequenz committed:
Update releases-notes with dependencies removed

Also fix the format of the release-notes.

Signed-off-by: Daniel Zullo <daniel.zullo@frequenz.com>

daniel-zullo-frequenz commented:
> Could you update release notes?

Sure, already done

leandro-lucarella-frequenz reviewed (approved):
I'm OK with leaving the removed dependencies from the release notes but in general it shouldn't be very relevant to users (except if there are some dependencies conflicts). All the dependencies are automatically installed by pip anyway. Any particular reason you asked for this @ela-kotulska-frequenz ?

ela-kotulska-frequenz commented:
It somehow influence users who plan to use panda.
For example their applications might stop working if they are using pandas, but they don't have pandas in their requirements list.

But I won't argue if you remove it.

leandro-lucarella-frequenz commented:
Let's merge as is. That would be a bug if they rely on a library they don't have in their dependencies, but I'm fine with being extra nice and letting them know that they should fix their bugs as we did with the SDK :smile:
